server/irmaserver/handle.go

```diff
@@ -104,6 +104,145 @@ func (session *session) handleGetStatus() (irma.ServerStatus, *irma.RemoteError)
+	// Create disclosure result instance
+	disclosureResult := &server.SessionDisclosureResult{
+		Token:  session.Result.Token,
+		Status: session.Result.ProofStatus,
+	}
+
+	// Depending on action add information
+	switch session.Result.Type {
+	case irma.ActionDisclosing:
+		request := session.request.(*irma.DisclosureRequest)
+		if request != nil {
+			disclosureResult.Identifier = request.Base().Identifier
+			disclosureResult.Nonce = request.GetNonce(nil)
+			disclosureResult.Requestor = server.SessionDisclosureResultRequestor{
+				Nonce: request.Base().Nonce,
+			}
+		}
+	case irma.ActionSigning:
+		disclosureResult.Nonce = session.Result.Signature.GetNonce()
+		request := session.request.(*irma.SignatureRequest)
+		if request != nil {
+			disclosureResult.Identifier = request.Base().Identifier
+			disclosureResult.Timestamp = session.Result.Signature.Timestamp
+			disclosureResult.Requestor = server.SessionDisclosureResultRequestor{
+				Message: &request.Message,
+				Nonce:   request.Base().Nonce,
+			}
+		}
+	default:
+		return nil, errors.New("Unknown result type")
+	}
+
+	// Array of credentials
+	var credentials []server.SessionDisclosureResultCredential
+
+	// Walk over proofs
+	for _, proof := range disclosure.Proofs {
+		credential := server.SessionDisclosureResultCredential{}
+		proofd, ok := proof.(*gabi.ProofD)
+
+		// Casting okay
+		if !ok {
+			return nil, errors.New("Unable to cast to proofd")
+		}
+
+		// Get meta data
+		metadata := irma.MetadataFromInt(proofd.ADisclosed[1], session.conf.IrmaConfiguration)
+		credType := metadata.CredentialType()
+
+		// Ensure the credential is valid
+		if credType == nil {
+			return nil, errors.New("Received unknown credential type")
+		}
+
+		// Ensure valid publickey
+		publickey, err := metadata.PublicKey()
+		if err != nil {
+			return nil, errors.New("Failed to parse public key")
+		}
+
+		// Set identifiers
+		credential.Identifier = credType.Identifier()
+		credential.Scheme = server.SessionDisclosureResultCredentialScheme{
+			Identifier:     credType.Identifier().SchemeManagerIdentifier(),
+			DistributedKey: session.conf.IrmaConfiguration.SchemeManagers[credType.Identifier().SchemeManagerIdentifier()].Distributed(),
+		}
+		credential.Issuer = server.SessionDisclosureResultCredentialIssuer{
+			Identifier: credType.Identifier().IssuerIdentifier(),
+			Publickey: server.SessionDisclosureResultCredentialIssuerPublickey{
+				Counter:    publickey.Counter,
+				ExpiryDate: publickey.ExpiryDate,
+				N:          publickey.N,
+				Z:          publickey.Z,
+				S:          publickey.S,
+				G:          publickey.G,
+				H:          publickey.H,
+				R:          publickey.R,
+			},
+		}
+
+		// Set credential information
+		credential.IssuedAt = metadata.SigningDate()
+		credential.ExpiresAt = metadata.Expiry()
+		credential.Proof = *proofd
+		credential.NotRevoked = proofd.NonRevocationProof != nil
```

Comment:
This means that just the presence of something within `proofd.NonRevocationProof`, regardless of whether or not that is a valid nonrevocation proof, will make the resulting datastructure claim that the credential is not revoked. The point of this data structure is to allow its user to validate its contents by itself, right? Why then make such a claim? Either the user wants to draw such conclusions itself, or not and instead trust the IRMA server's judgement - in which case it would probably not use this data structure and endpoint at all, and instead use the existing IRMA server API.

Reply:
Agreed, removed.